azurelinuxagent/common/cgroups.py

```diff
@@ -796,3 +778,27 @@ def set_memory_limit(self, limit=None, unit='megabytes'):
             fileutil.write_file(memory_limit_file, '{0}\n'.format(value))
         else:
             raise CGroupsException("Memory hierarchy not available in this cgroup")
+
+
+class CGroupsLimits(object):
+    def __init__(self, cgroup_name, threshold=None):
+        self.cpu_limit = threshold["cpu"] if threshold else self.get_default_cpu_limits(cgroup_name)
+        self.memory_limit = threshold["memory"] if threshold else self.get_default_cpu_limits(cgroup_name)
```

Comment:
Nope - you need `if threshold and "cpu" in threshold`. It should be legit to pass a hash that has a setting for only one value, thus asking for the default for the other value. If/when you add a third threshold, your first implementation task would be to add a new get_default call, but that's *all* you'd have to do; every existing use of a CGroupsLimits object would pass a hash that had no value for the new limit, and the `if threshold and "newlimit" in threshold` test would do the right thing.

Reply:
Good catch. Thanks.